exonum-java-binding/testkit/src/main/java/com.exonum.binding.testkit/TestKit.java

```diff
@@ -0,0 +1,197 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Collections;
+import javax.annotation.Nullable;
+
+/**
+ * TestKit for testing blockchain services. It offers simple network configuration emulation
+ * (with no real network setup). Although it is possible to add several validator nodes to this
+ * network, only one node (either validator or auditor) would be truly emulated and execute its
+ * {@link Service#afterCommit(BlockCommittedEvent)} method logic.
+ *
+ * @see <a href="https://exonum.com/doc/version/0.10/get-started/test-service/">TestKit documentation</a>
+ */
+public final class TestKit {
+
+  @VisibleForTesting
+  final static short MAX_SERVICE_NUMBER = 256;
+  private final static Injector frameworkInjector = Guice.createInjector(new TestKitFrameworkModule());
+
+  private final NativeHandle nativeHandle;
+  private final Map<Short, Service> services = new HashMap<>();
+
+  private TestKit(List<Class<? extends ServiceModule>> serviceModules, EmulatedNodeType nodeType,
+                  short validatorCount, @Nullable TimeProvider timeProvider) {
+    List<UserServiceAdapter> serviceAdapters = new ArrayList<>();
+    for (Class<? extends ServiceModule> moduleClass : serviceModules) {
+      UserServiceAdapter serviceAdapter = createUserModule(moduleClass);
+      serviceAdapters.add(serviceAdapter);
+      services.put(serviceAdapter.getId(), serviceAdapter.getService());
+    }
+    boolean isAuditorNode = nodeType == EmulatedNodeType.AUDITOR;
+    UserServiceAdapter[] userServiceAdapters = serviceAdapters.stream().toArray(UserServiceAdapter[]::new);
+    // TODO: fix after native implementation
+    nativeHandle = null;
+//    nativeHandle = new NativeHandle(
+//        nativeCreateTestKit(userServiceAdapters, isAuditorNode,
+//            validatorCount, timeProvider));
+  }
+
+  /**
+   * Creates a TestKit network with a single validator node for a single service.
+   */
+  public static TestKit forService(Class<? extends ServiceModule> serviceModule) {
+    return new TestKit(Collections.singletonList(serviceModule), EmulatedNodeType.VALIDATOR,
+        (short) 0, null);
+  }
+
+  @SuppressWarnings("unchecked")
+  public <T extends Service> T getService(short serviceId, Class<T> serviceClass) {
+    Service service = services.get(serviceId);
+    checkArgument(service.getClass().equals(serviceClass), "Service with id %s is not of expected class %s",
+        serviceId, serviceClass.getCanonicalName());
+    return (T) services.get(serviceId);
+  }
+
+  /**
+   * Creates a service from the service module.
+   *
+   * @param moduleClass a class of the user service module
+   */
+  private static UserServiceAdapter createUserModule(Class<? extends ServiceModule> moduleClass) {
+    try {
+      Constructor constructor = moduleClass.getDeclaredConstructor();
+      Object moduleObject = constructor.newInstance();
+      checkArgument(moduleObject instanceof Module, "%s is not a sub-class of %s",
```

Comment:
```suggestion
      checkArgument(moduleObject instanceof Module, "%s is not a sub-class of the %s",
```

Btw, looks like this check is always `true` because `ServiceModule` extends `Module`. No? I believe we can cast to `ServiceModule` at initialization and remove this check.

Reply:
This code seems to be copied from the place where we instantiate a class and subsequently a module _by name_, therefore, some checks can be redundant.

I'd also look into `multiple-services` branch. There might be something reusable here.

Reply:
`com.exonum.binding.runtime.ReflectiveModuleSupplier`